ts/live-compositor/src/types/registerInput.ts

```diff
@@ -0,0 +1,94 @@
+import * as Api from '../api';
+
+export type RegisterInput =
+  | ({ type: 'rtp_stream' } & RegisterRtpInput)
+  | ({ type: 'mp4' } & RegisterMp4Input);
+
+export type RegisterRtpInput = {
+  /**
+   * UDP port or port range on which the compositor should listen for the stream.
+   */
+  port: Api.PortOrPortRange;
+  /**
+   * Transport protocol.
+   */
+  transportProtocol?: Api.TransportProtocol | null;
+  /**
+   * Parameters of a video source included in the RTP stream.
+   */
+  video?: Api.InputRtpVideoOptions | null;
+  /**
+   * Parameters of an audio source included in the RTP stream.
+   */
+  audio?: InputRtpAudioOptions | null;
+  /**
+   * (**default=`false`**) If input is required and the stream is not delivered
+   * on time, then LiveCompositor will delay producing output frames.
+   */
+  required?: boolean | null;
+  /**
+   * Offset in milliseconds relative to the pipeline start (start request). If the offset is
+   * not defined then the stream will be synchronized based on the delivery time of the initial
+   * frames.
+   */
+  offsetMs?: number | null;
+};
+
+export type RegisterMp4Input = {
+  /**
+   * URL of the MP4 file.
+   */
+  url?: string | null;
+  /**
+   * Path to the MP4 file.
+   */
+  path?: string | null;
```

Comment:
maybe serverPath to avoid confusion if the node is deployed in a different location than the compositor.